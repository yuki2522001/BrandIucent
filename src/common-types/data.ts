import { Categories } from "./category";
import { Product } from "./product";

export interface DataContextProps {
  products: Product[];
  setProducts: Function;
  searchValue: [] | undefined;
  setSearchValue: Function;
  categories: Categories[];
  setCategories: Function;
}