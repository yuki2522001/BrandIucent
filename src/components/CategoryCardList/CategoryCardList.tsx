import React, { memo } from "react";
import CategoryList from "../CategoryList/CategoryList";

const CategoryCardList: React.FC = memo(() => {
  return (
    <div data-testid="categories" className="categories">
      <p className="categories__title">Categories</p>
      <CategoryList
        type="row"
        isSelect={true}
        onToggleCategory={() => {}}
        selectedCategories={[]}
        id={""}
      />
    </div>
  );
});

export default CategoryCardList;
