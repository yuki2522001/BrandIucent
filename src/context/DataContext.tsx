import { createContext, useMemo, useState } from "react";
import { Categories } from "@common-types/category";
import { DataContextProps } from "@common-types/data";
import { Product } from "@common-types/product";

const DataContext = createContext<DataContextProps>({} as DataContextProps);
const DataProvider: React.FC<{ children: JSX.Element[] | JSX.Element }> = ({
  children,
}) => {
  const [searchValue, setSearchValue] = useState<[]>([]);
  const [categories, setCategories] = useState<Categories[]>([]);
  const [products, setProducts] = useState<Product[]>([]);

  const value = {
    products,
    setProducts,
    searchValue,
    setSearchValue,
    categories,
    setCategories,
  };

  return <DataContext.Provider value={value}>{children}</DataContext.Provider>;
};

export { DataProvider, DataContext };
