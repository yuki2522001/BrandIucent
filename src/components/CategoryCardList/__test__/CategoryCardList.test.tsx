import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { DataContextProps } from "@common-types/data";
import { DataContext } from "@context/DataContext";
import "@testing-library/jest-dom";
import CategoryCardList from "../CategoryCardList";

const contextProductMock: DataContextProps = {
  searchValue: [],
  setSearchValue: jest.fn(),
  categories: [],
  setCategories: jest.fn(),
  products: [],
  setProducts: jest.fn(),
};

describe("Categories component", () => {
  test("should render categories component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <DataContext.Provider value={contextProductMock}>
        <Router location={history.location} navigator={history}>
          <CategoryCardList />
        </Router>
      </DataContext.Provider>
    );
    expect(getByTestId("categories")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <DataContext.Provider value={contextProductMock}>
        <Router location={history.location} navigator={history}>
          <CategoryCardList />
        </Router>
      </DataContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
