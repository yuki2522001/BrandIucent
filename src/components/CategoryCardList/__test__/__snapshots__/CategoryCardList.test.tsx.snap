// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="categories"
    data-testid="categories"
  >
    <p
      class="categories__title"
    >
      Categories
    </p>
    <div
      class="categoryList__row"
      data-testid="category-list"
    />
  </div>
</DocumentFragment>
`;
